### DISCUSSION TITLE
Problems with attachments from inbound emails

### DISCUSSION THREAD

**Discussion by trainor1** [django-helpdesk/django-helpdesk#1107] [Q&A] [answered]
Hi All,

I am having 2 problems with attachments on inbound emails:

1. I cannot attach .zip files and have them placed in the ticket.  The error message that I am getting in the log file is:  ['Unsupported file extension: .zip'].  My major use case is helping students in Python programming courses.  In my previous ticket system, there was no problem with students zipping up their PyCharm projects and uploading them to a ticket.  So, this is an important use case for me.   I can't see a place to configure allowed file extensions. Can you point me to a solution or a work-around?

2. Other documents (like .pdf files) upload just fine.  There are links appearing in my tickets.  Unfortunately, clicking on those links gets me a 404 page.  I am suspecting that the solution to this problem is to configure the HTTP server to serve files from the attachments directory itself and not pass these requests to the Django server.  Can you confirm that for me.  Can you think of a way to get these links to work when using the development server?  It would nice to be able to test the links during configuration/testing and not in production.

Thanks for your help.

Best,

Kevin

**Comment by brucegibbins** [comment]
Hello

Just a quick reply as I am away from PC.

There is a helpdesk settings.py entry to extend the allowed attachment
extensions. If you take a look at the online documentation I believe it is
mentioned. There are also a number of closed issues in GitHub where this
dame question has been raised.

There has been a recent rewrite of mail attachment handling which has been
pushed to GitHub as a new branch so it can be tested. I will hopefully be
able to do my own testing of that branch next week as I am hoping it will
handle attachments better.

**Reply by trainor1** [reply]
Hi Bruce,

I scoured the documentation and every imaginable settings file in the distribution.  This setting is not mentioned.

As your promised, I did find a closed issue that was on point:  Zip file as an attachment is required. #1057.  This says that information about the setting will be added to the documentation.  But, I don't believe that the documentation change has been made.  Or, perhaps it has been made and it has not yet been published.

I will be interested in learning more about your new code for attachments when you have finished your work.

**Reply by brucegibbins** [reply]
All the work is @uhurusurfa . I'm just about ready to test it against O365

**Comment by brucegibbins** [comment]
Also the attachments are stored in the MEDIA Directory. Again this is
defined in settings.py but may not be there. There is plenty if django doco
on how to set this up and server it.

Cheers

**Reply by trainor1** [reply]
Thanks for the MEDIA settings reminder.  I did not have that properly set.

**Comment by uhurusurfa** [comment, marked as answer]
Check this post for how to enable ZIP files:
https://github.com/django-helpdesk/django-helpdesk/issues/1057

**Reply by trainor1** [reply]
Thanks for the tip.  I had already found my way to that post.  It solved my problem.

**Reply by brucegibbins** [reply]
Hi. @uhurusurfa 

I noticed that settings.rst has the VALID_EXTENSIONS setting mentioned but that isn't on the [](https://django-helpdesk.readthedocs.io/en/0.3/settings.html ) documentation. Perhaps you could shed some light for me on how to refresh that? or what the protocol is as it is in the branch for the email parsing. Cheers

**Reply by uhurusurfa** [reply]
There is now an automation set up in readthedocs to automatically build the docs when a new release tag is created.
The VAID_EXTENSIONS entry is currently not released - it is in the branch I pushed up wit hthe email parsing enhancements so will be released to readthedocs when we merge that and make a new release.

**Reply by brucegibbins** [reply]
Look at testing thebemail parsing soon. Had a few issues at my end with machine rebuild. Sorry for delay.
